MapleServer2/Types/BaseStats.cs

```diff
@@ -322,6 +322,66 @@ public static long CriticalRate(JobCode jobCode, short level)
         };
     }
 
+    public static long SpiritRegen(JobCode jobCode)
+    {
+        return jobCode switch
+        {
+            JobCode.Beginner => 1,
+            JobCode.Knight => 0,
+            JobCode.Berserker => 1,
+            JobCode.Wizard => 1,
+            JobCode.Priest => 8,
+            JobCode.Archer => 7,
+            JobCode.HeavyGunner => 0,
+            JobCode.Thief => 1,
+            JobCode.Assassin => 1,
+            JobCode.Runeblade => 1,
+            JobCode.Striker => 1,
+            JobCode.SoulBinder => 1,
+            _ => 0,
+        };
+    }
+
+    public static long SpiritRegenInterval(JobCode jobCode)
+    {
+        return jobCode switch
+        {
+            JobCode.Beginner => 100,
+            JobCode.Knight => 300,
+            JobCode.Berserker => 100,
+            JobCode.Wizard => 100,
+            JobCode.Priest => 500,
+            JobCode.Archer => 500,
+            JobCode.HeavyGunner => 0,
+            JobCode.Thief => 100,
+            JobCode.Assassin => 100,
+            JobCode.Runeblade => 100,
+            JobCode.Striker => 100,
+            JobCode.SoulBinder => 1000,
+            _ => 0,
+        };
+    }
+
+    public static float SpiritRegenIntervalRate(JobCode jobCode)
+    {
+        return jobCode switch
+        {
+            JobCode.Beginner => 1,
+            JobCode.Knight => 1,
+            JobCode.Berserker => 1,
+            JobCode.Wizard => 2,
+            JobCode.Priest => 1,
+            JobCode.Archer => 1,
+            JobCode.HeavyGunner => 1,
+            JobCode.Thief => 1,
+            JobCode.Assassin => 1,
+            JobCode.Runeblade => 2,
+            JobCode.Striker => 1,
+            JobCode.SoulBinder => 1,
+            _ => 35,
+        };
+    }
+
```

Comment:
Maybe these values would be nicer if it was in Constants.

Reply:
ConstantStats.cs or Constants.json?

Reply:
neither, lol. we should probably put them together.
I meant MapleServer2/Constants/Constant.cs

Reply:
Oh that one slipped past my solution search. I'll put it there then